syntax = "proto3";

message ComResponse {
    int32 code = 1;  
}

enum ErrorCodeAstro {
    NO_ERROR = 0; // No Error
    WS_PARSE_PROTOBUF_ERROR = -1; // Protobuf parsing failed
    WS_SDCARD_NOT_EXIST = -2; // SD card not detected
    WS_INVALID_PARAM = -3; // Invalid parameter
    WS_SDCARD_WRITE_ERROR = -4; // Image writing to SD card failed (maybe the card is full)
    CODE_ASTRO_PLATE_SOLVING_FAILED = -11500; // Plate solving failed
    CODE_ASTRO_FUNCTION_BUSY = -11501; // Astronomical function busy
    CODE_ASTRO_CALIBRATION_FAILED = -11504; // Calibration failed
    CODE_STEP_MOTOR_LIMIT_POSITION_WARNING = -14518; // GOTO limit warning
    CODE_STEP_MOTOR_LIMIT_POSITION_HITTED = -14519; // Star collision limit
    CODE_ASTRO_GOTO_FAILED = -11505; // GOTO failed
    CODE_ASTRO_NEED_GOTO = -11513; // No GOTO
    CODE_SYSTEM_SET_TIME_FAILED = -13300; // Set time failed
    CODE_SYSTEM_SET_TIMEZONE_FAILED = -13301; // Failed to set time zone
}

enum AstroState {
    ASTRO_STATE_IDLE = 0; // Idle state
    ASTRO_STATE_RUNNING = 1; // Running
    ASTRO_STATE_STOPPING = 2; // Is stopping
    ASTRO_STATE_STOPPED = 3; // Has stopped
    ASTRO_STATE_PLATE_SOLVING = 4; // Astronomy is Plating Solving
}

message ResNotifyStateAstroGoto {
    AstroState state = 1;  
}

message ResNotifyHostSlaveMode {
    int32 mode = 1;
}

message ResNotifyStateAstroCalibration {
    AstroState state = 1;  
    int32 plate_solving_times = 2;
}

message ResNotifyStateAstroTracking {
    AstroState state = 1;  
    string target_name = 2;
}

enum AstroTrackingSpecial {
    TRACKING_SUN = 0;
    TRACKING_MOON = 1;
}

message ResNotifyStateAstroTrackingSpecial {
    AstroState state = 1;  
    string target_name = 2;
    int32 index = 3;
}